cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/co/cask/cdap/datapipeline/SmartWorkflow.java

```diff
@@ -165,6 +168,18 @@ protected void configure() {
 
   @Override
   public void initialize(WorkflowContext context) throws Exception {
+    MDC.put(Constants.PIPELINE_LIFECYCLE_TAG, Constants.PIPELINE_LIFECYCLE_TAG_VALUE);
```

Comment:
I think it's better to have a wrapper Logger class that takes a normal Logger, mdc key, and mdc value in its constructor, and wraps every logging method with the MDC put and remove operations. It's a little more work now, but more re-usable.